tests/dragonfly/conftest.py

```diff
@@ -28,6 +28,30 @@
 logging.getLogger("asyncio").setLevel(logging.WARNING)
 
 DATABASE_INDEX = 0
+BASE_LOG_DIR = "/tmp/dragonfly_logs/"
+
+
+def pytest_configure(config):
+    if os.path.exists(BASE_LOG_DIR):
+        shutil.rmtree(BASE_LOG_DIR)
+
+
+def pytest_runtest_setup(item):
+    # Generate a unique directory name for each test based on its nodeid
+    translator = str.maketrans(":[]{}/ ", "_______", "\"*'")
+    unique_dir = item.nodeid.translate(translator)
+    test_dir = os.path.join(BASE_LOG_DIR, unique_dir)
+    if os.path.exists(test_dir):
+        shutil.rmtree(test_dir)
+    os.makedirs(test_dir)
```

Comment:
This generates annoyingly long folder names. For example on the latest test failure: 

```
🪵🪵🪵🪵🪵🪵 /tmp/dragonfly_logs/dragonfly_cluster_test.py__test_cluster_fuzzymigration_df_seeder_factory0-df_factory0-3-16-20000_/dragonfly.3719f2c306a1.root.log.WARNING.20240708-144404.33601.log 🪵🪵🪵🪵🪵🪵
```

ideally it should just be the name of the logfile:

```
🪵🪵🪵🪵🪵🪵 dragonfly.3719f2c306a1.root.log.WARNING.20240708-144404.33601.log 🪵🪵🪵🪵🪵🪵
```

Reply:
Also this dir name contains test parameters that can be useful

Reply:
The problem with the long name is that it gives us headache when we need to find the logs. Look how compact it was before https://github.com/dragonflydb/dragonfly/actions/runs/9816081863/job/27105881487#step:6:931

Why would we replace this with something verbose?

> Also this dir name contains test parameters that can be useful 

When a test fails it also contains the parameters and the name of the tests so this is redundant

see https://github.com/dragonflydb/dragonfly/actions/runs/9816081863/job/27105881487#step:6:553

Reply:
It helps you to see everything in one place

Reply:
I agree that the folder name is a bit long. Check out if you can simplify it with only the test name and params i.e 
/tmp/dragonfly_logs/test_cluster_fuzzymigration_df_seeder_factory0-df_factory0-3-16-20000/
but I also find this helpfull to have the tests name as the folder name.
This way I dont need to find the log name in the ui if the test failed though I do agree that most cases we have only one test failing there are cases where we have several and in this cases it will be eazier to find the log by the test name
Also when running pytests on my laptop I can check the logs easily this way and also will know how to check the logs of non failed tests